### ISSUE TITLE
Action creators from `createSlice` are not inferred as `PayloadActionCreator`

### ISSUE THREAD

**Issue by markerikson** [reduxjs/redux-starter-kit#157]
I'm finally getting a chance to use RSK myself in a real project, and I just ran into an interesting issue.

In `createAction()`, we specifically attach the generated type to the generated action creator so that we can use the action creator itself as the type value, by both overriding `toString()` and adding it as `actionCreator.type = type`.

I've got some code I just wrote that looks like:

```ts
const someSlice = createSlice({
  /* omit */
});

const {doSomething} = someSlice.actions;

// later, in a middleware
switch(action.type) {
    case doSomething.type: { // error: Property 'type' does not exist on type '(payload: SomePayload) => PayloadAction<SomePayload, string>'

    }
}
```

So, even though we're using `createAction()` inside of `createSlice`, there's a disconnect in the types.

In our TS code, we specifically have [a `PayloadActionCreator` type](https://github.com/reduxjs/redux-starter-kit/blob/2eb5b4bf49a44e64757e54d64d9593e7ce7d74da/src/createAction.ts#L18-L22) that appears to define itself as "a function with this payload, plus the `type` field".

However, I don't see that `PayloadActionCreator` type being used in `createSlice` anywhere.  Instead, there's [a `SliceActionCreator` type](https://github.com/reduxjs/redux-starter-kit/blob/2eb5b4bf49a44e64757e54d64d9593e7ce7d74da/src/createSlice.ts#L13-L15) that reuses the `PayloadAction` definition.

Seems like `SliceActionCreator` ought to be referring to `PayloadActionCreator` somehow?

**Comment by markerikson** [comment]
/cc @denisw, @jessidhia, @Dudeonyx , @phryneas :  

any ideas?

**Comment by markerikson** [comment]
I just tried specifically hand-editing the typings for `SliceActionCreator` in my local app codebase `node_modules` to be:

```ts
export declare type SliceActionCreator<P> = P extends void ? PayloadActionCreator<void> : PayloadActionCreator<P>;
```
And that seems to compile okay.

**Comment by phryneas** [comment]
Those two ActionCreators seem to have diverged on the zero-argument-case - even though they actually are exactly the same internally. 
Drives me nuts since forever, actually :) (See my comment here: https://github.com/reduxjs/redux-starter-kit/pull/149#issuecomment-504729239)

I guess I'll try to make those behave similarly when I touch them for #149 anyways :)

**Comment by markerikson** [comment]
Meh.  I tried to poke around at this some myself right now, but don't have time since I'm at work.

What I'm seeing is that my suggested change breaks the expected test failure in `createSlice.types.ts`:

```ts
  // typings:expect-error
  counter.actions.multiply()
```

Which I think is verifying that if the reducer is defined as _not_ taking an action arg, that the action creator rejects attempts to call it without an argument.

I also added this test, which fails without with the change but passes with it:

```ts
  const incrementType = counter.actions.increment.type
```

@phryneas, could you do me a favor and try to resolve just this one issue for now and pick up with the payload customization thing later?

**Comment by phryneas** [comment]
> Which I think is verifying that if the reducer is defined as not taking an action arg, that the action creator rejects attempts to call it without an argument.

I guess adding this behaviour to the `PayloadActionCreator` would be okay? I guess it's what's intended anyways.

> @phryneas, could you do me a favor and try to resolve just this one issue for now and pick up with the payload customization thing later?

Sure thing :)

**Comment by markerikson** [comment]
Fixed via #158 .
